p2p/chain.py

```diff
@@ -378,6 +325,35 @@ def _skip_empty_and_duplicated_receipts(
             # as a warning so we don't forget about it.
             self.logger.warn("Got unexpected msg: %s (%s)", cmd, msg)
 
+    async def _handle_block_receipts(
+            self, peer: ETHPeer, receipts: List[List[eth.Receipt]]) -> None:
+        self.logger.debug("Got Receipts for %d blocks from %s", len(receipts), peer)
+        loop = asyncio.get_event_loop()
+        iterator = map(make_trie_root_and_nodes, receipts)
+        receipts_tries = await wait_with_token(
+            loop.run_in_executor(None, list, iterator),
+            token=self.cancel_token)
+        receipt_roots = []
+        for receipt_root, trie_dict_data in receipts_tries:
+            await self.chaindb.coro_persist_trie_data_dict(trie_dict_data)
+            receipt_roots.append(receipt_root)
```

Comment:
What do you think about changing this to the following (after the exit of the loop)

```python
receipt_roots = tuple(zip(*receipt_tries))[0]
```

Reply:
I'd prefer the current version as it doesn't have any cognitive overhead and when debugging one can immediately see the list will contain all receipt roots (and just them) at the end of the loop, whereas the latter requires me to remember/lookup how zip works, and in fact I wasn't even aware it is a variadic function

Reply:
Ok, how about:

```python
receipt_roots = [receipt_root for receipt_root, _ in receipt_tries]
```

Simply removes the loop based accumulation in favor of a comprehension which maintains readability.

Reply:
Just an FYI so you know it exists, I just found [`cytoolz.pluck`](http://toolz.readthedocs.io/en/latest/api.html#toolz.itertoolz.pluck) 

```python
receipt_roots = tuple(pluck(0, receipt_tries))  # cast to tuple to force evaluation
```